unlock-app/src/components/interface/locks/List/elements/LockCard.tsx

```diff
@@ -140,33 +174,42 @@ const LockIcon = ({ lock }: LockIconProps) => {
   )
 }
 
-export const LockCard = ({ lock, network, isLoading }: LockCardProps) => {
+export const LockCard = ({ lock, network }: LockCardProps) => {
   const { networks } = useConfig()
+  const web3service = useWeb3Service()
   const lockAddress = lock.address
   const [isCopied, setCopied] = useClipboard(lockAddress, {
     successDuration: 2000,
   })
-
   const { explorer, baseCurrencySymbol } = networks?.[network] ?? {}
 
   const explorerUrl = explorer?.urls?.address(lockAddress) || '#'
 
   const symbol = (lock as any)?.currencySymbol ?? baseCurrencySymbol
 
+  const getLockDetail = async () => {
+    return await web3service.getLock(lock.address, network)
+  }
+
+  const {
+    isLoading: isLoadingInfo,
+    data: { keyPrice, balance, outstandingKeys, expirationDuration } = {},
+  } = useQuery(['lockDetail', network, lock?.address], async () =>
+    getLockDetail()
+  )
+
```

Comment:
we actually have `balance` and `outstandingKeys ` values from the subgraph

Reply:
Until the value is not present we show a placeholder 

<img width="1411" alt="Screenshot 2022-10-26 at 12 44 11" src="https://user-images.githubusercontent.com/20865711/198006640-3806b15c-178a-4b2b-901d-aa9b7ac6130f.png">

Reply:
Ok lets actually make sure we add these to the subgraphs before merging.
@kalidiagne I think that could be a great occasion to learn about how to update the subgraph.
Maybe you can schedule a sessions with @clemsos so he can show you?

Reply:
Sure ill do it! i take first look at the example @clemsos already provided, and the talk also with him!

Reply:
Since we dont have the `balance` in the graph and it's not easy to add, just use the `getBalance` function to get it...